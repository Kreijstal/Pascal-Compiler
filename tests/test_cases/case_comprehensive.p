program CaseComprehensive;
var
  x: integer;
begin
  { Test 1: match first case }
  x := 1;
  case x of
    1: writeln(100);
    2: writeln(200);
    3: writeln(300)
  end;
  
  { Test 2: match middle case }
  x := 2;
  case x of
    1: writeln(100);
    2: writeln(200);
    3: writeln(300)
  end;
  
  { Test 3: match last case }
  x := 3;
  case x of
    1: writeln(100);
    2: writeln(200);
    3: writeln(300)
  end;
  
  { Test 4: no match, with else }
  x := 5;
  case x of
    1: writeln(100);
    2: writeln(200);
    3: writeln(300)
  else
    writeln(999)
  end;
  
  { Test 5: match with compound statement }
  x := 2;
  case x of
    1: writeln(100);
    2: begin
         writeln(201);
         writeln(202)
       end;
    3: writeln(300)
  end;
end.
