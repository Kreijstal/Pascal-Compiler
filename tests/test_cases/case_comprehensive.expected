100
200
300
999
201
202
